core/src/test/scala/integration/kafka/server/KRaftClusterTest.scala

```diff
@@ -778,4 +779,52 @@ class KRaftClusterTest {
       cluster.close()
     }
   }
+  def createAdminClient(cluster: KafkaClusterTestKit, useController: Boolean): Admin = {
+    var props: Properties = null
+    props = cluster.clientProperties()
+    props.put(AdminClientConfig.CLIENT_ID_CONFIG, this.getClass.getName)
+    Admin.create(props)
+  }
+
+  @Test
+  def testDescribeQuorumRequestToBrokers() : Unit = {
+    val cluster = new KafkaClusterTestKit.Builder(
+      new TestKitNodes.Builder().
+        setNumBrokerNodes(4).
+        setNumControllerNodes(3).build()).build()
+    try {
+      cluster.format
+      cluster.startup
+      for (i <- 0 to 3) {
+        TestUtils.waitUntilTrue(() => cluster.brokers.get(i).brokerState == BrokerState.RUNNING,
+          "Broker Never started up")
+      }
+      val admin = createAdminClient(cluster, false)
+      try {
+        val quorumState = admin.describeMetadataQuorum(new DescribeMetadataQuorumOptions)
+        val quorumInfo = quorumState.quorumInfo.get()
+
+        assertEquals(0, quorumInfo.observers.size)
+        assertEquals(3, quorumInfo.voters.size)
```

Comment:
A more straightforward way to assert the voter/observer sets is in `DescribeQuorumRequestTest`:
```scala
      val voterData = partitionData.currentVoters.asScala
      assertEquals(cluster.controllerIds().asScala, voterData.map(_.replicaId).toSet);

      val observerData = partitionData.observers.asScala
      assertEquals(cluster.brokerIds().asScala, observerData.map(_.replicaId).toSet);
```

Reply:
That check already exists in the `DescribeQuorumRequestTest`. The count validation here is just a sanity check.

Reply:
We are checking the size and all of the individual replicaIds below. For example:
```
assertTrue(-1 < observer.replicaId && 4 > observer.replicaId,
            s"Observer ID ${observer.replicaId} was not within expected range.")
 ```
I am just offering a more concise and complete way to do that.

Reply:
You may have missed my comment above. The nice thing about asserting the set directly is that it ensures that all expected voters/observers are present in the result.